x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_editor.tsx

```diff
@@ -306,24 +299,28 @@ export function DimensionEditor(props: DimensionEditorProps) {
               incompatibleSelectedOperationType={incompatibleSelectedOperationType}
               onDeleteColumn={() => {
                 setState(
-                  deleteColumn({
+                  mergeLayer({
                     state,
                     layerId,
-                    columnId,
+                    newLayer: deleteColumn({ layer: state.layers[layerId], columnId }),
                   })
                 );
               }}
               onChoose={(choice) => {
-                let column: IndexPatternColumn;
+                let newLayer: IndexPatternLayer;
                 if (
                   !incompatibleSelectedOperationType &&
                   selectedColumn &&
                   'field' in choice &&
                   choice.operationType === selectedColumn.operationType
                 ) {
-                  // If we just changed the field are not in an error state and the operation didn't change,
-                  // we use the operations onFieldChange method to calculate the new column.
-                  column = changeField(selectedColumn, currentIndexPattern, fieldMap[choice.field]);
+                  newLayer = replaceColumn({
+                    layer: state.layers[layerId],
+                    columnId,
+                    indexPattern: currentIndexPattern,
+                    field: fieldMap[choice.field],
+                    // TODO: don't keep params?
+                  });
                 } else {
                   // Otherwise we'll use the buildColumn method to calculate a new column
```

Comment:
This comment is not understandable without the other one (line 324) that was just removed. Could you update it? The information can still be valid, but eg. the method name changed.